web3js-ext/src/web3/web3.ts

```diff
@@ -28,7 +45,21 @@ export class KlaytnWeb3 extends Web3 {
     this.eth.sendSignedTransaction = this.eth_sendSignedTransaction(this);
     
     // TODO: Connect web3.klay, web3.net, etc from @klaytn/web3rpc
+    this.openApiClient = makeApiClient(provider.clientUrl);
```

Comment:
In Web3, the connection options (such as HTTP Headers and Basic Auth) are found inside HttpProvider.httpProviderOptions as "RequestInit" type. See
- https://docs.web3js.org/guides/web3_providers_guide/#provider-options
- https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
- https://github.com/klaytn/caver-js-examples/blob/main/transaction/send_klay/index.js#L48-L57

So this line should be something like:

```ts
if (this.provider instanceof HttpProvider) {
  // Access to private properties
  const url = (this.provider as any).clientUrl; 
  const opts = (this.provider as any).httpProviderOptions as RequestInit;

  const openApiClient = makeApiClient(url, opts);
  this.admin = new AsyncOpenApi(new AdminApi(openApiClient));
  // ...
}
// otherwise API methods are not created.
```

Reply:
@nohkwak Please test against [KAS API](https://www.klaytnapi.com/ko/resource/docs/tutorial/node/node-api/) which requires you to specify custom HTTP headers.

```js
const auth = "Basic: " + Buffer.from(accessKeyId + ':' + secretAccessKey).toString('base64');
const option1 = {
  headers: {
    "Authorization": auth,
    "x-chain-id": "1001",
  }
}
const option2 = {
  headers: [
    ["Authorization", auth],
    ["x-chain-id", value: "1001"],
  ]
}
const option3 = {
  headers: [
    {name: "Authorization", value: auth},
    {name: "x-chain-id", value: "1001"},
  ]
}
const provider = new web3.providers.HttpProvider("https://node-api.klaytnapi.com/v1/klaytn", option1);
const web3 = new KlaytnWeb3(provider);

await web3.klay.blockNumber();
```